packages/common/src/hooks/useImageSize.ts

```diff
@@ -82,77 +97,121 @@ export const useImageSize = <
   onDemand,
   size,
   sizes
-}: UseImageSizeProps<ImageSize, ImageSizes>) => {
+}: BaseUserImageSizeProps<ImageSize, ImageSizes> & {
+  onDemand?: boolean
+}): Maybe<string> | (() => Maybe<string>) {
   const [getPreviousId, setPreviousId] = useInstanceVar<number | null>(null)
 
-  const fallbackImage = (url: URL) => {
-    setPreviousId(null)
-    return url
-  }
-
-  const getSmallerImage = getNextImage<ImageSize, ImageSizes>((a, b) => a < b)
-  const getLargerImage = getNextImage<ImageSize, ImageSizes>((a, b) => a > b)
+  const getSmallerImage = useMemo(
+    () => getNextImage<ImageSize, ImageSizes>((a, b) => a < b),
+    []
+  )
+  const getLargerImage = useMemo(
+    () => getNextImage<ImageSize, ImageSizes>((a, b) => a > b),
+    []
+  )
 
-  const getImageSize = (): Maybe<URL> => {
+  const getImageSize = useCallback((): { url: Maybe<URL>; type: string } => {
     if (id === null || id === undefined || typeof id === 'string') {
-      return ''
+      return { url: '', type: 'empty' }
+    }
+
+    const fallbackImage = (url: URL) => {
+      setPreviousId(null)
+      return url
     }
 
     // No image sizes object
     if (!sizes) {
-      return fallbackImage('')
+      return { url: fallbackImage(''), type: 'none' }
     }
 
     // An override exists
     if (DefaultSizes.OVERRIDE in sizes) {
       const override: Maybe<URL> = sizes[DefaultSizes.OVERRIDE]
       if (override) {
-        return fallbackImage(override)
+        return { url: fallbackImage(override), type: 'override' }
       }
 
-      return defaultImage
+      return { url: defaultImage, type: 'default' }
     }
 
     // The desired size exists
     if (size in sizes) {
       const desired: Maybe<URL> = sizes[size]
       if (desired) {
-        return desired
+        return { url: desired, type: 'desired' }
       }
 
-      return defaultImage
+      return { url: defaultImage, type: 'default' }
     }
 
     // A larger size exists
     const larger: Maybe<URL> = getLargerImage(sizes, size)
     if (larger) {
-      return fallbackImage(larger)
-    }
-
-    // If no larger size exists, dispatch to get the desired size
-    // Don't dispatch twice for the same id
-    if (load && getPreviousId() !== id) {
-      setPreviousId(id)
-      // Request the desired size
-      dispatch(action(id, size))
```

Comment:
This was the issue, since this was technically happening in the render cycle

Reply:
ooooooouyyydddddffff

i'm sure i wrote this